### ISSUE TITLE
Potential example for the docs: how to transform any deterministic regression model to a probabilistic binary classification model, and then evaluate it with k-fold cross validation

### ISSUE THREAD

**Issue by DilumAluthge** [JuliaAI/MLJBase.jl#760]
This gives the full code for taking any deterministic single-label regression model, transforming it into a probabilistic single-label binary-class classification model, and evaluating it with k-fold cross validation with a metric (area under the ROC curve) that is used for evaluating single-label binary-class classification models.

The specific model that we wrap is the `EvoTreeRegressor` model from the `EvoTrees` package, with the `loss` set to `:logistic`. The code works on both CPUs and GPUs.

@ablaom @OkonSamuel I figure I would just stick this in an issue for now, but maybe in the future we can find somewhere in the documentation/examples/tutorials to put it.

```julia
import EvoTrees
import MLJ

const EvoTreeRegressor = MLJ.@load EvoTreeRegressor pkg=EvoTrees

function transform(
        as_categorical::AbstractVector;
        levels,
        pool,
    )
    negative_class = levels[1]
    positive_class = levels[2]
    n = length(as_categorical)
    as_numerical = Vector{Float64}(undef, n)
    for i in 1:n
        cat = as_categorical[i]
        if cat == positive_class
            num = Float64(1)
        elseif cat == negative_class
            num = Float64(0)
        else
            msg = "Invalid value: $(cat)"
            throw(ErrorException(msg))
        end
        as_numerical[i] = num
    end
    return as_numerical
end

function inverse_transform(
        as_single_probability::AbstractVector;
        levels,
        pool,
    )
    negative_class = levels[1]
    positive_class = levels[2]
    n = length(as_single_probability)
    as_probability_distribution = Vector{MLJ.UnivariateFinite}(undef, n)
    for i in 1:n
        prob = as_single_probability[i]
        pd = MLJ.UnivariateFinite(
            levels,
            prob;
            pool,
            augment = true
        )
        as_probability_distribution[i] = pd
    end
    return as_probability_distribution
end

X, y = MLJ.@load_crabs

positive_class = "B"
negative_class = "O"

levels = [negative_class, positive_class]
pool = MLJ.CategoricalPool(levels)

device = "cpu"
# device = "gpu"

loss = :logistic
regression_model = EvoTreeRegressor(;
    device,
    loss,
)

transformed_model = MLJ.TransformedTargetModel(
    regression_model;
    target  = v -> transform(v;         levels, pool),
    inverse = v -> inverse_transform(v; levels, pool),
)

# This will print some scitype warnings, which are expected, because the underlying
# model is a deterministic regression model, but we have categorical labels.
mach = MLJ.machine(
    transformed_model,
    X,
    y;
    # scitype_check_level = 2, # TODO: uncomment this line
)

nfolds = 5
resampling = MLJ.CV(; nfolds, shuffle = true)
measure = MLJ.auc

# In this case, we must specify the operation; otherwise, we get an error.
# We also need to set `check_measure` to `false` to avoid another error.
performance_evaluation = MLJ.evaluate!(
    mach;
    resampling,
    measure,
    operation = MLJ.predict,
    check_measure = false,
)
```

**Comment by DilumAluthge** [comment]
Here's example output from the CPU:

```
Evaluating over 5 folds: 100%[=========================] Time: 0:00:43
PerformanceEvaluation object with these fields:
  measure, measurement, operation, per_fold,
  per_observation, fitted_params_per_fold,
  report_per_fold, train_test_pairs
Extract:
┌──────────────────┬─────────────┬───────────┬────────────────────────────────────┐
│ measure          │ measurement │ operation │ per_fold                           │
├──────────────────┼─────────────┼───────────┼────────────────────────────────────┤
│ AreaUnderCurve() │ 0.905       │ predict   │ [0.917, 0.95, 0.818, 0.909, 0.932] │
└──────────────────┴─────────────┴───────────┴────────────────────────────────────┘
```

**Comment by ablaom** [comment]
@DilumAluthge Thanks for this. I think ultimately we should have a wrapper, but this is great to have documented.

**Comment by EngrStudent** [comment]
This is nice.  It is like a high quality CV answer, but without a question.  
What is the question?  What made the effort a good investment of time?
Is this an example, like for example code?  Is this a feature request in that you provide python and it gets translated?
